cro-clib/src/jsonrpc.rs

```diff
@@ -0,0 +1,181 @@
+) -> Result<AppWalletClient> {
+    let signer_manager = WalletSignerManager::new(storage.clone());
+    let transaction_cipher = get_tx_query(tendermint_client.clone())?;
+    let transaction_builder = DefaultWalletTransactionBuilder::new(
+        signer_manager,
+        tendermint_client.genesis().unwrap().fee_policy(),
+        transaction_cipher,
+    );
+    Ok(DefaultWalletClient::new(
+        storage,
+        tendermint_client,
+        transaction_builder,
+        Some(50),
+    ))
+}
+
+fn make_ops_client(
+    storage: SledStorage,
+    tendermint_client: WebsocketRpcClient,
+) -> Result<AppOpsClient> {
+    let transaction_cipher = get_tx_query(tendermint_client.clone())?;
+    let signer_manager = WalletSignerManager::new(storage.clone());
+    let fee_algorithm = tendermint_client.genesis().unwrap().fee_policy();
+    let wallet_client = make_wallet_client(storage, tendermint_client.clone())?;
+    Ok(DefaultNetworkOpsClient::new(
+        wallet_client,
+        signer_manager,
+        tendermint_client,
+        fee_algorithm,
+        transaction_cipher,
+    ))
+}
+
+fn make_syncer_config(
+    storage: SledStorage,
+    tendermint_client: WebsocketRpcClient,
+) -> Result<AppSyncerConfig> {
+    let transaction_cipher = get_tx_query(tendermint_client.clone())?;
+
+    Ok(AppSyncerConfig::new(
+        storage,
+        tendermint_client,
+        transaction_cipher,
+        true,
+        50,
+        50,
+    ))
+}
+
+fn do_jsonrpc_call(
+    storage_dir: &str,
+    websocket_url: &str,
+    network_id: u8,
+    json_request: &str,
+) -> Result<String> {
+    let mut io = IoHandler::new();
+    let storage = SledStorage::new(storage_dir)?;
+    let tendermint_client = WebsocketRpcClient::new(websocket_url)?;
+    let wallet_client = make_wallet_client(storage.clone(), tendermint_client.clone())?;
+    let ops_client = make_ops_client(storage.clone(), tendermint_client.clone())?;
+    let syncer_config = make_syncer_config(storage, tendermint_client)?;
+
+    let multisig_rpc = MultiSigRpcImpl::new(wallet_client.clone());
+    let transaction_rpc = TransactionRpcImpl::new(network_id);
+    let staking_rpc = StakingRpcImpl::new(wallet_client.clone(), ops_client, network_id);
+    let sync_rpc = SyncRpcImpl::new(syncer_config);
+    let wallet_rpc = WalletRpcImpl::new(wallet_client, network_id);
+
+    io.extend_with(multisig_rpc.to_delegate());
+    io.extend_with(transaction_rpc.to_delegate());
+    io.extend_with(staking_rpc.to_delegate());
+    io.extend_with(sync_rpc.to_delegate());
+    io.extend_with(wallet_rpc.to_delegate());
+
+    // let request = r#"{"jsonrpc": "2.0", "method": "say_hello", "params": [42, 23], "id": 1}"#;
+    io.handle_request_sync(json_request)
+        .err_kind(ErrorKind::InvalidInput, || {
+            "stateful command execute failed"
+        })
+}
+
+/// # Safety
+///
+/// Should not be called with null pointers.
+#[no_mangle]
+pub unsafe extern "C" fn jsonrpc_call(
```

Comment:
this makes temporary rpc server,
so in this context, io port , should be found by using socket api (bind)

but rpc server everytime is very slow.
i suggest separating into two pars
1. initialize rpc server  <- initialize_json_rpc
2. use rpc server via rpc <- json_rpc_call

Reply:
No server gets started, directly calls the RPC functions, not go through the socket.